odl/contrib/fom/supervised/supervised_fom.py

```diff
@@ -0,0 +1,570 @@
+# Copyright 2014-2017 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+"""
+Implementation of Figure of Merits (FOMs) for comparing reconstructions with
+a given reference.
+"""
+
+__all__ = ('mean_square_error', 'mean_absolute_error', 'mean_value_difference',
+           'standard_deviation_difference', 'range_difference', 'blurring',
+           'false_structures')
+
+
+def mean_square_error(test_data, ref_data, mask=None, normalized=False):
+    """Mean square error-based FOM between ``test_data`` and ``ref_data``.
+
+    Evaluates mean square error between input (``test_data``) and
+    reference data (``ref_data``), by measuring consistency using the L2 -norm.
+    Allows for normalization (``normalized``) and a masking of the two spaces
+    (``mask``).
+
+    Notes
+    ----------
+    The FOM evaluates
+
+    .. math::
+        \| f - g \|^2_2,
+
+    or, in normalized form
+
+    .. math::
+        \\frac{\| f - g \|^2_2}{\| f \|^2_2 + \| g \|^2_2}.
+
+    The normalized FOM takes its values in [0, 1], with higher correspondance
+    at lower FOM value.
+
+    Parameters
+    ----------
+    test_data : `LinearSpaceElementElement`
+        Input data or reconstruction.
+    ref_data : `LinearSpaceElement`
+        Reference data to compare ``test_data`` to.
+    mask : `LinearSpaceElement`
+        Binary mask to define ROI in which FOM evaluation is performed.
+    normalized  : `Boolean`
```

Comment:
\`Boolean\` -> bool

(Explanation: Single-tick keywords are tried to match to names defined by us, that can be classes, functions or glossary terms. That only applies to things like \`Operator\` or \`LinearSpace\` but not to built-in types like `bool`, `int`, `float`, `list`, `tuple` etc.. For those we don't make links because there would be just too many.)